mlrun/track/tracker_manager.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+from .platform_trackers import MLFlowTracker
+from .tracker import Tracker
+from mlrun.execution import MLClientCtx
+from os import environ
+
+
+class TrackerManager:
+    def __init__(self):
+        self._trackers = []
+
+    def add_tracker(self, tracker: Tracker):
+        """
+        adds a Tracker to tracking list
+        """
+        self._trackers.append(tracker())
+
+    def pre_run(
+        self, context: MLClientCtx, mode: str = None, env: dict = None
+    ) -> (dict, dict):
+        """
+        goes over all trackers and calls there pre_run function
+
+         Returns:
+            two dictionaries, env and args containing environment and tracking data for all trackers
+        """
+        run_env = env or {}
+        kwargs = {"mode": mode}
+        for tracker in self._trackers:
+            if tracker.is_enabled():
+                env = tracker.pre_run(context, run_env, kwargs)
+                environ.update(env)  # needed in order to set mlflow experiment id
```

Comment:
comment is for mlflow yet it should be generic

Reply:
done